packages/loader/src/ktx2/KTX2Loader.ts

```diff
@@ -104,10 +102,21 @@ export class KTX2Loader extends Loader<Texture2D | TextureCube> {
   override initialize(engine: Engine, configuration: EngineConfiguration): Promise<void> {
     if (configuration.ktx2Loader) {
       const options = configuration.ktx2Loader;
-      if (
-        // @ts-ignore
-        KTX2Loader._detectSupportedFormat(engine._hardwareRenderer, options.formatPriorities) === KTX2TargetFormat.ASTC
-      ) {
+      const priorityFormats = options.priorityFormats;
+      const supportedList = new Array<KTX2TargetFormat>();
+      if (Array.isArray(priorityFormats[0])) {
+        for (let i = 0; i < priorityFormats.length; i++) {
+          supportedList.push(
+            KTX2Loader._detectSupportedFormat((engine as any)._hardwareRenderer, <KTX2TargetFormat[]>priorityFormats[i])
+          );
+        }
+      } else {
+        supportedList.push(
+          KTX2Loader._detectSupportedFormat((engine as any)._hardwareRenderer, <KTX2TargetFormat[]>priorityFormats)
+        );
+      }
+
+      if (supportedList.every((format) => KhronosTranscoder.transcoderMap[format])) {
```

Comment:
Your `load` should use the same logic with this, maybe you should encapsulation a function